pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -519,6 +529,15 @@ func partitionCacheExtents(req MetricsQueryRequest, extents []Extent, minCacheEx
 		requests = append(requests, req)
 	}
 
+	// Track samples processed from cache
+	var cachedSamplesProcessed uint64 = 0
+	for _, stepStat := range cachedPerStepStat {
+		cachedSamplesProcessed += uint64(stepStat.Value)
+	}
+	if details := QueryDetailsFromContext(ctx); details != nil {
+		details.SamplesProcessedFromCache = cachedSamplesProcessed
```

Comment:
I'm a bit confused. `partitionCacheExtents()` can be called multiple times for a single input query. Here we overwrite `SamplesProcessedFromCache` each time 🤔

Reply:
Yep, my bad; thanks for catching it.  Pushed a fix